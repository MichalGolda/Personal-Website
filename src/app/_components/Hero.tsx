import { PrimaryButton, SecondaryOutlinedButton } from "./Button";

export default function Hero() {
  return (
    <div className="text-center items-center flex flex-col">
      <h1 className="text-secondary font-bold text-6xl">
        Buduje spersonalizowane <br />
        <span className="text-primary">strony internetowe</span>,{" "}
        <span className="text-primary">aplikacje webowe</span>
      </h1>
      <p className="text-2xl text-body font-normal max-w-3xl mt-8">
        Koduję świat pełen możliwości - tworząc rozwiązania, wpisuję swoją pasję
        w cyfrową rzeczywistość.
      </p>
      <div className="space-x-8 mt-10">
        <PrimaryButton className="w-64">Stwórzmy coś razem</PrimaryButton>
        <SecondaryOutlinedButton className="w-64">
          Portfolio
        </SecondaryOutlinedButton>
      </div>
    </div>
  );
}
