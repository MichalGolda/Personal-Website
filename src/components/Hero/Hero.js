import {
  StyledHero,
  StyledTitle,
  StyledDescription,
  StyledTextContainer,
  StyledImage,
  StyledImageContainer,
} from "./Hero.styled";
import Button from "../Button/Button";

export default function Hero() {
  return (
    <StyledHero>
      <StyledTextContainer>
        <StyledTitle>fullstack developer</StyledTitle>
        <StyledDescription>
          Vivamus posuere velit in ullamcorper rhoncus. Aenean odio eros,
          iaculis at tincidunt id, feugiat ac nisl. Pellentesque volutpat.
        </StyledDescription>
        <a href="#contact">
          <Button variant="primary">Kontakt</Button>
        </a>
      </StyledTextContainer>
      <StyledImageContainer>
        <StyledImage src="hero-illustration.svg" alt="hero illustration" fill />
      </StyledImageContainer>
    </StyledHero>
  );
}
