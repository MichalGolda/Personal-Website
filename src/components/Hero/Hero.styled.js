import styled from "styled-components";
import Image from "next/image";

export const StyledHero = styled.div`
  display: flex;
  row-gap: 16px;
  justify-content: left;
  flex-direction: column;
  margin-top: -48px;

  @media (${({ theme }) => theme.breakpoints.md}) {
    flex-direction: row;
    justify-content: space-between;
  }
`;

export const StyledTitle = styled.h1`
  font-size: 3rem;
  font-weight: 900;
  line-height: 100.5%;
  text-transform: uppercase;

  @media (${({ theme }) => theme.breakpoints.md}) {
    font-size: 4rem;
  }
`;

export const StyledDescription = styled.p`
  font-size: 1.25rem;
  color: ${({ theme }) => theme.colors.secondary};
`;

export const StyledTextContainer = styled.div`
  display: flex;
  row-gap: 1rem;
  flex-direction: column;
`;

export const StyledImageContainer = styled.div`
  min-width: 420px;
  display: none;

  @media (${({ theme }) => theme.breakpoints.lg}) {
    display: initial;
  }
`;

export const StyledImage = styled(Image)`
  position: inherit !important;
  display: none;

  @media (${({ theme }) => theme.breakpoints.md}) {
    display: inherit;
  }
`;
