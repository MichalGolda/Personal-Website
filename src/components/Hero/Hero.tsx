import {
  StyledHero,
  StyledTitle,
  StyledDescription,
  StyledTextContainer,
  StyledImage,
  StyledImageContainer,
} from "./Hero.styled";
import LinkButton from "../LinkButton/LinkButton";

export default function Hero() {
  return (
    <StyledHero>
      <StyledTextContainer>
        <StyledTitle>Fullstack Developer</StyledTitle>
        <StyledDescription>
          Cześć! Mam na imię Michał, od kilku lat interesuje się programowaniem
          i tworzeniem stron internetowych. Do realizacji projektów wykorzystuje
          technologie oraz techniki, które pozwalają zachować najwyższą jakość
          świadczonych usług.
        </StyledDescription>
        <LinkButton href="#contact" variant="primary">
          Kontakt
        </LinkButton>
      </StyledTextContainer>
      <StyledImageContainer>
        <StyledImage src="hero-illustration.svg" alt="hero illustration" fill />
      </StyledImageContainer>
    </StyledHero>
  );
}
