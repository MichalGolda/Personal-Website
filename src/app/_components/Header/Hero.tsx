import { PrimaryButton, SecondaryOutlinedButton } from "@/app/_components";

export default function Hero() {
  return (
    <div className="flex items-center justify-center h-[calc(100%-32px)] w-full -mt-8">
      <div className="text-center items-center flex flex-col">
        <h1 className="text-secondary font-bold text-6xl">
          Buduje spersonalizowane <br />
          <span className="text-primary">strony internetowe</span>,{" "}
          <span className="text-primary">aplikacje webowe</span>
        </h1>
        <p className="text-2xl text-body font-normal max-w-3xl mt-8">
          Koduję świat pełen możliwości - tworząc rozwiązania, wpisuję swoją
          pasję w cyfrową rzeczywistość.
        </p>
        <div className="space-x-8 mt-10">
          <a href="#contact">
            <PrimaryButton className="w-64">Stwórzmy coś razem</PrimaryButton>
          </a>
          <a href="#portfolio">
            <SecondaryOutlinedButton className="w-64">
              Portfolio
            </SecondaryOutlinedButton>
          </a>
        </div>
      </div>
    </div>
  );
}
