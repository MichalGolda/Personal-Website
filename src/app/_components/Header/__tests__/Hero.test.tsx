import "@testing-library/jest-dom"
import { render, screen } from "@testing-library/react"
import Hero from "../Hero"

describe("Hero component", () => {
    it("renders without errors", () => {
        render(<Hero />)

        const title = screen.getByTestId("title")
        const subtitle = screen.getByTestId("subtitle")
        const links = screen.getAllByRole("link")
        const contactLink = links[0]
        const contactButton = contactLink.children[0]
        const portfolioLink = links[1]
        const portfolioButton = portfolioLink.children[0]

        expect(title).toBeInTheDocument()
        expect(title.tagName).toBe("H1")

        expect(subtitle).toBeInTheDocument()
        expect(subtitle.tagName).toBe("P")
    
        expect(contactLink).toBeInTheDocument()
        expect(contactLink).toHaveAttribute("href", "#contact")

        expect(contactButton.textContent).toBe("Stwórzmy coś razem")

        expect(portfolioLink).toBeInTheDocument()
        expect(portfolioLink).toHaveAttribute("href", "#portfolio")
        expect(portfolioLink.textContent).toBe("Portfolio")
    })

    it("renders unchanged", () => {
        const { container } = render(<Hero />)

        expect(container).toMatchSnapshot()
    })
})