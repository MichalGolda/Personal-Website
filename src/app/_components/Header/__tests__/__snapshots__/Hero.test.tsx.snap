// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero component renders unchanged 1`] = `
<div>
  <div
    class="flex items-center justify-center h-[calc(100%-61.8px)] md:h-[calc(100%-93.8px)] w-full md:-mt-8"
  >
    <div
      class="text-center items-center flex flex-col"
    >
      <h1
        class="text-4xl md:text-5xl lg:text-6xl text-secondary font-bold"
        data-testid="title"
      >
        Buduje spersonalizowane 
        <br />
        <span
          class="text-primary"
        >
          strony internetowe
        </span>
        ,
         
        <span
          class="text-primary"
        >
          aplikacje webowe
        </span>
      </h1>
      <p
        class="text-lg md:text-xl lg:text-2xl text-body font-normal max-w-3xl mt-4 md:mt-8"
        data-testid="subtitle"
      >
        Koduję świat pełen możliwości - tworząc rozwiązania, wpisuję swoją pasję w cyfrową rzeczywistość.
      </p>
      <div
        class="flex flex-col gap-4 md:flex-row mt-8 md:mt-10"
      >
        <a
          href="#contact"
        >
          <button
            class="py-4 px-8 font-bold text-sm md:text-base transition-all duration-500 disabled:bg-lightGrey disabled:text-body bg-primary text-white hover:bg-primary/90 active:bg-primary/80 border border-transparent border-solid w-64"
          >
            Stwórzmy coś razem
          </button>
        </a>
        <a
          href="#portfolio"
        >
          <button
            class="py-4 px-8 font-bold text-sm md:text-base transition-all duration-500 disabled:bg-lightGrey disabled:text-body bg-transparent border-secondary border border-solid hover:bg-secondary active:bg-secondary/80 hover:text-white text-secondary w-64"
          >
            Portfolio
          </button>
        </a>
      </div>
    </div>
  </div>
</div>
`;
